R/config.R

```diff
@@ -336,6 +336,8 @@ python_config <- function(python, required_module, python_versions, forced = NUL
       python_libdir <- config[[var]]
       ext <- switch(Sys.info()[["sysname"]], Darwin = ".dylib", Windows = ".dll", ".so")
       libpython <- file.path(python_libdir, paste0("libpython" , version, c("", "m"), ext))
+      # look for any unconventional library extensions (e.g., in Centos 7)
+      libpython <- c(libpython, unlist(lapply(libpython, function(x) list.files(path = dirname(x), pattern = paste0(basename(x), "*"), full.names = TRUE))))
```

Comment:
I think this would be simpler as:

``` r
pattern <- paste("^libpython", version, "m?", ext)
libpython <- list.files(python_libdir, pattern = pattern, full.names = TRUE)
```

Reply:
> pattern <- paste("^libpython", version, "m?", ext)

Agree. Change the above to `paste0` and it's perfect!

Reply:
I've updated this request as per @kevinushey suggestion. I had to add several other changes so that we only need `list.files()` to find the Python library files. As the new `for` loop, it's actually not that necessary but can simplify the nested `if`/`else`.

If this is too much I can revert back to only change two lines from my initial pull request :-)